File: leetcode/python/2610.py
class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        hashmap = {}
        r = []

        for n in nums:
            if n not in hashmap:
                hashmap[n] = 1
            else:
                hashmap[n] += 1

            if len(r) < hashmap[n]:
                r.append([n])
            else:
                r[hashmap[n] - 1].append(n)
        return r

File: leetcode/rust/2610.rs
use std::collections::HashMap;

impl Solution {
    pub fn find_matrix(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let mut res = Vec::new();
        let mut hash_map = HashMap::new();

        for &n in &nums {
            if hash_map.get(&n).unwrap_or(&0) == &0 {
                hash_map.insert(n, 1);
            } else {
                hash_map.insert(n, hash_map[&n] + 1);
            }

            if res.len() < hash_map[&n] {
                res.push(vec![n]);
            } else {
                res[hash_map.get(&n).unwrap() - 1].push(n);
            }
        }

        return res;
    }
}
